src/pages/commits/ActiveCommits/buildVariantCard/WaterfallTaskStatusIcon.test.tsx

```diff
@@ -53,6 +56,29 @@ describe("waterfallTaskStatusIcon", () => {
       );
     });
   });
+
+  it("should call the appropriate functions on hover and unhover", async () => {
+    (injectGlobalStyle as jest.Mock).mockImplementationOnce(
+      (taskIdentifier: string) => {
+        Promise.resolve(taskIdentifier);
+      }
+    );
+    (removeGlobalStyle as jest.Mock).mockImplementationOnce(() => {});
+
+    const { queryByDataCy } = render(Content("failed"), {
+      route: "/commits/evergreen",
+    });
+    userEvent.hover(queryByDataCy("waterfall-task-status-icon"));
+    await waitFor(() => {
+      expect(injectGlobalStyle).toHaveBeenCalledTimes(1);
+      expect(injectGlobalStyle).toHaveBeenCalledWith(props.identifier);
+    });
+
+    userEvent.unhover(queryByDataCy("waterfall-task-status-icon"));
+    await waitFor(() => {
+      expect(removeGlobalStyle).toHaveBeenCalledTimes(1);
+    });
```

Comment:
Should we assert that they had some styles applied?

Reply:
I'm going to forgo this because as far as I can tell the styles on the element itself doesn't change (maybe some magic happening?). I could check that the document head is updated, but since I do that in the other test file anyways, it'd probably be unnecessary to do it twice